package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -651,16 +651,18 @@ def _get_dh_pairs(self):
 
         return donors, hydrogens
 
-    def _filter_atoms(self, donors, hydrogens, acceptors):
-        """Filter donor, hydrogen and acceptor atoms to consider only hydrogen
-        bonds between two or more specified groups.
+    def _filter_atoms(self, donors, acceptors):
+        """Create a mask to filter donor, hydrogen and acceptor atoms.
+
+        This can be used to consider only hydrogen bonds between two or more
+        specified groups.
 
         Groups are specified with the `between` keyword when creating the
         HydrogenBondAnalysis object.
 
            Returns
            -------
-           donors, hydrogens, acceptors: Filtered AtomGroups
+           mask: np.ndarray
```

Comment:
This is a private method so there's no problem changing the return value when we need to. Given that you already updated the docs, let's also do a versionchanged, just for internal use:
```suggestion
           mask: np.ndarray


        .. versionchanged:: 2.5.0
           Change return value to a mask instead of separate AtomGroups.
``